### PR TITLE
Nova custom component

### PR DESCRIPTION
**Pull request by Sina-pz**
Angular material:
mat-slider-bar
mat-slider-toggle
mat-checkbox
mat-radio-button

### CODE REVIEW

**Review by dhaval2989 — APPROVED** [review]
good job

File: src/app/reusable/nv-custom-component/nv-custom-component.component.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<ng-container [ngSwitch]="component">
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-custom-component/nv-custom-component.component.html:1]: I didn't understand what do you mean my NV-custom-component, every reusable components is a custom component

File: src/app/reusable/nv-custom-component/nv-custom-component.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+@Component({
+  selector: 'app-nv-custom-component',
+  templateUrl: './nv-custom-component.component.html',
+  styleUrls: ['./nv-custom-component.component.scss']
+})
+export class NvCustomComponentComponent implements OnInit {
+  @Input() component!: NvCustomComponent;
+  nvCustomComponent = NvCustomComponent;
+// checkBox
+checked = false;
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-custom-component/nv-custom-component.component.ts:18]: don't declare value directly without the datatype to the variable

File: src/app/reusable/nv-custom-component/nv-custom-component.component.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+<mat-checkbox *ngSwitchCase="nvCustomComponent.CheckBox" [(ngModel)]="checked" [(indeterminate)]="indeterminate"
```
Comment (veeranjaneyuludavuluri) [inline src/app/reusable/nv-custom-component/nv-custom-component.component.html:4]: As discussed previously you created a reusable nova checkbox and slidebar but why did you put everything in single component, I hope this is not a good practice

**Review by dhaval2989 — APPROVED** [review]
good to go.

File: src/app/reusable/nv-custom-component/nv-custom-component.component.html (hunk 3 above)
Comment (Sina-pz) [reply src/app/reusable/nv-custom-component/nv-custom-component.component.html:4]: I hope you could remember your idea!!!!!!!!
[Tuesday 2:18 PM] Sina Pashazanous
    See angular has radio button, checkk-box, slide, and etc
    do you think we need to have separate component for each of them?

[Tuesday 2:20 PM] Veeranjaneyulu Davuluri
    Let's create a custom nova component and put Angular components into it

File: src/app/reusable/nv-custom-component/nv-custom-component.component.html (hunk 1 above)
Comment (Sina-pz) [reply src/app/reusable/nv-custom-component/nv-custom-component.component.html:1]: I hope you could remember your idea!!!!!!!!
[Tuesday 2:18 PM] Sina Pashazanous
    See angular has radio button, checkk-box, slide, and etc
    do you think we need to have separate component for each of them?

[Tuesday 2:20 PM] Veeranjaneyulu Davuluri
    Let's create a custom nova component and put Angular components into it

File: src/app/reusable/nv-custom-component/nv-custom-component.component.ts (hunk 2 above)
Comment (Sina-pz) [reply src/app/reusable/nv-custom-component/nv-custom-component.component.ts:18]: when we initialize, we are defining the type as well
 if we do both we are doing repetitive job

checked = false;
(property) NvCustomComponentComponent.checked: boolean (Angular understand it)

File: src/app/reusable/nv-custom-component/nv-custom-component.component.html (hunk 1 above)
Comment (veeranjaneyuludavuluri) [reply src/app/reusable/nv-custom-component/nv-custom-component.component.html:1]: I mean to say, for check-box, we need to create a nova-checkbox
I didn't mean that component name should be custom component and put everything over there

File: src/app/reusable/nv-custom-component/nv-custom-component.component.ts (hunk 2 above)
Comment (veeranjaneyuludavuluri) [reply src/app/reusable/nv-custom-component/nv-custom-component.component.ts:18]: I am not taking about input directive variables, there are some other variable like, checked, autoticks and many

File: src/app/reusable/nv-custom-component/nv-custom-component.component.html (hunk 3 above)
Comment (veeranjaneyuludavuluri) [reply src/app/reusable/nv-custom-component/nv-custom-component.component.html:4]: same

File: src/app/reusable/nv-custom-component/nv-custom-component.component.html (hunk 1 above)
Comment (Sina-pz) [reply src/app/reusable/nv-custom-component/nv-custom-component.component.html:1]: I asked you is it ok to do So, but you wanted me to "Let's create a custom nova component and put Angular components into it" but now again you changed your words!!
see I do not have time for that I will address it after my tasks would get done

My question: See angular has radio button, checkk-box, slide, and etc
do you think we need to have separate component for each of them? =>>> like nova-checkbox, 

your answer: Let's create a custom nova component and put Angular components into it !!!

**Review by veeranjaneyuludavuluri — APPROVED** [review]
Good to go

**Pull request merged**